src/stub/core_impl/base.rs

```diff
@@ -238,18 +238,19 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
 
                     let addr = addr + NumCast::from(i).ok_or(Error::TargetMismatch)?;
                     let data = &mut buf[..chunk_size];
-                    match target.base_ops() {
+                    let data_len = match target.base_ops() {
                         BaseOps::SingleThread(ops) => ops.read_addrs(addr, data),
                         BaseOps::MultiThread(ops) => {
                             ops.read_addrs(addr, data, self.current_mem_tid)
                         }
                     }
                     .handle_error()?;
+                    debug_assert!(data_len <= data.len());
 
                     n -= chunk_size;
                     i += chunk_size;
 
-                    res.write_hex_buf(data)?;
+                    res.write_hex_buf(&data[..data_len])?;
```

Comment:
one key goal of `gdbstub` is to be panic free, which means this code needs to be resilient against misbehaving target implementations that return a longer than expected length.

that means no asserts/debug_asserts, and no direct indexing (unless the bounds check has been optimized out)

The way this is currently done throughout the codebase is by doing something like

```rust
// TODO: add more specific error variant?
data.get(..data_len).ok_or(Error::PacketBufferOverflow)?
```

If you grep for that error type around the codebase, you'll find many comments along the lines of `// TODO: add more specific error variant?`, since it's not _really_ appropriate to recycle this error for this use case.

...buuuut that's a bigger fish to fry. for this PR, just do what other code is doing (including that same comment), and I might take a pass at fixing it as part of tackling #112